cloud/client.go

```diff
@@ -59,6 +59,7 @@ type Client struct {
 	installationName         string
 	logstreamAddressOverride string
 	serverConnTimeout        time.Duration
+	orgIDCache               map[string]string // orgName -> orgID
```

Comment:
one quick check: is there any chance this will be accessed concurrently? `sync.Map` is not terribly hard to use and would prevent any data races in such a case. I doubt the performance impact is notable for our purposes. It might be worth using here.

Reply:
good point! Fixed.